apps/site/assets/ts/projects/components/Banner.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from "react";
+import { ProjectTeaserWithDate, Route } from "./__projects";
+
+interface Props {
+  banner: ProjectTeaserWithDate;
+}
+
+const bannerRouteClass = ({ routes }: ProjectTeaserWithDate): string =>
+  cmsRouteToClass(routes[0]);
+
+const bannerBgClass = (banner: ProjectTeaserWithDate): string => {
+  if (banner.routes.length === 0) return "u-bg--unknown";
+  return bannerRouteClass(banner);
+};
+
+const toCSSClass = (name: string): string =>
+  name
+    .toLowerCase()
+    .replace(" ", "-")
```

Comment:
We should use a regex here, because according to [the api docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace), "If `pattern` is a string, only the first occurrence will be replaced." And since we have to use a regex anyway, might as well combine both calls into one:

```
name
  .toLowerCase()
  .replace(/[ _]/g, "-")
```

Reply:
I took this from https://github.com/mbta/dotcom/blob/master/apps/site/assets/ts/stop/components/RoutePillList.tsx#L15

Since there was extra CMS specific logic for groups I didn't take the whole function. There shouldn't really be a route that has more than one hyphen but I can change it in both places.